crates/ruff_linter/src/rules/ruff/rules/sort_dunder_all.rs

```diff
@@ -0,0 +1,495 @@
+use std::cmp::Ordering;
+
+use ruff_diagnostics::{Applicability, Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast as ast;
+use ruff_python_ast::whitespace::indentation;
+use ruff_python_codegen::Stylist;
+use ruff_python_parser::{lexer, Mode, Tok};
+use ruff_source_file::Locator;
+use ruff_text_size::{Ranged, TextRange, TextSize};
+
+use crate::checkers::ast::Checker;
+
+use itertools::Itertools;
+
+/// ## What it does
+/// Checks for `__all__` definitions that are not alphabetically sorted.
+///
+/// ## Why is this bad?
+/// Consistency is good. Use a common convention for `__all__` to make your
+/// code more readable and idiomatic.
+///
+/// ## Example
+/// ```python
+/// import sys
+///
+/// __all__ = [
+///     "b",
+///     "c",
+///     "a",
+/// ]
+///
+/// if sys.platform == "win32":
+///     __all__ += ["z", "y"]
+/// ```
+///
+/// Use instead:
+/// ```python
+/// import sys
+///
+/// __all__ = [
+///     "a",
+///     "b",
+///     "c",
+/// ]
+///
+/// if sys.platform == "win32":
+///     __all__ += ["y", "z"]
+/// ```
+///
+/// ## Fix safety
+/// This rule's fix should be safe for single-line `__all__` definitions
+/// and for multiline `__all__` definitions without comments.
+/// For multiline `__all__` definitions that include comments,
+/// the fix is marked as unsafe, as it can be hard to tell where the comments
+/// should be moved to when sorting the contents of `__all__`.
```

Comment:
This is an intereseting use of unsafe, but it fits `unsafe`'s definition.

> The fix may be what the user intended, but it is uncertain; the resulting code will have valid syntax.

I would probably keep the fix as safe if we feel like it does the right thing in the majority of cases even if there are comments. But I'm also okay going wiht unsafe (can also be easily changed later). Interested to hear what other people think.